roofit/roofitcore/inc/RooRatio.h

```diff
@@ -0,0 +1,59 @@
+// Author: Rahul Balasubramanian, Nikhef 01 Apr 2021
+
+/*****************************************************************************
+ * RooFit
+ * Authors:                                                                  *
+ *   WV, Wouter Verkerke, UC Santa Barbara, verkerke@slac.stanford.edu       *
+ *   DK, David Kirkby,    UC Irvine,         dkirkby@uci.edu                 *
+ *                                                                           *
+ * Copyright (c) 2000-2019, Regents of the University of California          *
+ *                          and Stanford University. All rights reserved.    *
+ *                                                                           *
+ * Redistribution and use in source and binary forms,                        *
+ * with or without modification, are permitted according to the terms        *
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)             *
+ *****************************************************************************/
+
+#ifndef ROO_RATIO
+#define ROO_RATIO
+
+#include "RooAbsReal.h"
+#include "RooProduct.h"
+#include "RooRealProxy.h"
+
+#include <list>
+
+class RooRealVar;
+class RooArgList;
+class RooProduct;
+
+class RooRatio : public RooAbsReal {
+public:
+  RooRatio();
+  RooRatio(const char *name, const char *title, RooAbsReal &numerator,
```

Comment:
As I have recently learned that RooFactoryWSTool employs the constructors in the order in which they appear in the header file, I would suggest we reorder these lines as follows:
double,double
double,RooAbsReal
RooAbsReal,double
RooAbsReal,RooAbsReal
RooArgList,RooArgList

Also, I would vote in favour of using RooAbsCollection rather than RooArgList, as the order is not important, so it could be a RooArgSet as well.

Reply:
Hi, that's a good point ! the constructors are reordered appropriately. I hesitate a bit about the rooabscollection over rooarglist, this is anyway passed on to a rooproduct that takes a rooarglist so do we need this option ?

Reply:
I have never understood why RooProduct uses a RooArgList anyway, maybe we can ping Wouter on this. Perhaps we can add a RooProduct constructor that takes a RooArgSet or change that to RooAbsCollection as well?

Reply:
BTW, as this is a bit of a tangent, I've created an issue for the design flaws in RooProduct and RooProdPdf: https://github.com/root-project/root/issues/7809

Reply:
Yes, we should follow this up with Wouter. There are some examples [0] where I see the use case of RooArgSet being passed to the RooProduct, so it is unclear to me if we need to have different constructors. Given that there is going to be some latency in discussing this, it would be best to investigate this in a separate PR.

[0] - https://github.com/root-
project/root/blob/f6d1543b44fb3ab959b5b5b9d73df7fb61361237/tutorials/roofit/rf704_amplitudefit.C#L51

Reply:
That's a good point! I assume you don't propose to make these changes to `RooProduct` in this PR, right? I can take care of this once this PR is merged so you can continue focusing on integrating your EFT morphing developments.

Reply:
agreed, that was a bit the point of the separate issue, to make sure that we can merge this now and it doesn't get lost.
we just need to remember that we probably want to harmonize RooRatio alongside RooProduct and RooProdPdf in that separate PR, but this can be added to the issue and we can move along with this PR here.